internal/server/statvar/fetcher/formula_fetcher.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fetcher
+
+import (
+	"context"
+	"net/http"
+	"sort"
+
+	pb "github.com/datacommonsorg/mixer/internal/proto"
+	pbv2 "github.com/datacommonsorg/mixer/internal/proto/v2"
+	"github.com/datacommonsorg/mixer/internal/server/resource"
+	v1pv "github.com/datacommonsorg/mixer/internal/server/v1/propertyvalues"
+	"github.com/datacommonsorg/mixer/internal/store"
+	"github.com/datacommonsorg/mixer/internal/util"
+	"golang.org/x/sync/errgroup"
+)
+
+// FetchFormulas fetches StatisticalCalculations from storage and returns a map of SV dcids to a list of inputPropertyExpressions.
+func FetchFormulas(
+	ctx context.Context,
+	store *store.Store,
+	metadata *resource.Metadata,
+) (map[string][]string, error) {
+	errGroup, errCtx := errgroup.WithContext(ctx)
+	localRespChan := make(chan map[string]map[string]map[string][]*pb.EntityInfo, 1)
+	remoteRespChan := make(chan *pbv2.NodeResponse, 1)
+	// Fetch for BT and SQL.
+	errGroup.Go(func() error {
+		data, _, err := v1pv.Fetch(
+			errCtx,
+			store,
+			[]string{"StatisticalCalculation"},
+			[]string{"typeOf"},
+			0,
+			"",
+			"in",
+		)
+		if err != nil {
+			return err
+		}
+		statisticalCalculations := []string{}
+		for _, nodes := range data["StatisticalCalculation"]["typeOf"] {
+			for _, node := range nodes {
+				statisticalCalculations = append(statisticalCalculations, node.Dcid)
+			}
+		}
+		localResp, _, err := v1pv.Fetch(
+			errCtx,
+			store,
+			statisticalCalculations,
+			[]string{"outputProperty", "inputPropertyExpression"},
+			0,
+			"",
+			"out",
+		)
+		if err != nil {
+			return err
+		}
+		localRespChan <- localResp
+		return nil
+	})
+	// Fetch for Remote Mixer.
+	if metadata.RemoteMixerDomain != "" {
+		errGroup.Go(func() error {
+			req := &pbv2.NodeRequest{
+				Nodes:    []string{"StatisticalCalculation"},
+				Property: "<-typeOf",
+			}
+			statisticalCalculationResp := &pbv2.NodeResponse{}
```

Comment:
nit: golang uses short var names, also i found it easy to read :)

Given the very long lines in this function, maybe use naming like statCalResp, statCal, outputProps, inputProps, outputExp, inputExp etc?

Reply:
Good idea thanks! Done